### PR TITLE
stride pooling for max and average layer

### LINKED ISSUE #1608: Feature Request: Sequence Pooling with Stride
Problem:

- Currently, sequence [Pooling](http://www.paddlepaddle.org/doc/ui/api/trainer_config_helpers/poolings.html) in PaddlePaddle is performed sequence-wise. After pooling, a sequence is transformed into a non-sequence.
- This is not enough  in building a deep one-dimensional convolution model, such as models for NLP or speech tasks, in which pooling layer may be followed by a new convolution layer. 
- According to current design, such a model is impossible, because output of pooling is no longer a sequence, that cannot be input to a new convolution layer.

**It will be helpful if the sequence Pooling layer provides a stride parameter, which controls  the scope of pooling operation**. 

It may work like this:
- If stride is set -1, it equals to sequence-wise pooling as it does now, otherwise, pooling is performed upon a small local area.
- After pooling with stride n (n is smaller than sequence length), a long sequence will be shorten.

### PR DESCRIPTION
**Pull request by luotao1**
fix #1608.
finish feature of "Sequence Pooling with Stride".

### CODE REVIEW

File: python/paddle/trainer_config_helpers/layers.py [hunk 1]
```diff
@@ -1112,6 +1118,8 @@ def pooling_layer(input,
     :param pooling_type: Type of pooling, MaxPooling(default), AvgPooling,
                          SumPooling, SquareRootNPooling.
     :type pooling_type: BasePoolingType|None
+    :param stride: window size.
```
Comment (lcy-seso) [inline python/paddle/trainer_config_helpers/layers.py]: stride --> the step size between successive pooling regions.

File: python/paddle/trainer_config_helpers/layers.py [hunk 2]
```diff
@@ -1090,10 +1090,16 @@ def pooling_layer(input,
...
                   agg_level=AggregateLevel.TO_NO_SEQUENCE,
+                  stride=-1,
                   layer_attr=None):
     """
     Pooling layer for sequence inputs, not used for Image.
 
+    If stride > 0, this layer slides a window whose size is determined by stride,
+    and return the pooling value of the window as the output. Thus, a long sequence
+    will be shorten. Note that for sequence with sub-sequence, the default value
+    of stride is -1.
```
Comment (lcy-seso) [inline python/paddle/trainer_config_helpers/layers.py]: the parameter stride specifies the intervals at which to apply the pooling operation.

Comment (luotao1) [inline python/paddle/trainer_config_helpers/layers.py]: Done

File: python/paddle/trainer_config_helpers/layers.py (hunk 1 above)
Comment (luotao1) [inline python/paddle/trainer_config_helpers/layers.py]: Done, and update the comments in last_seq and first_seq at the same time.

File: paddle/gserver/layers/MaxLayer.h [hunk 3]
```diff
@@ -26,6 +26,11 @@ namespace paddle {
  * If SequenceLevel = kNonSeq:
  *    Output: output size is the number of input sequences (NOT input instances)
  *    output[i] = max_{for each instance in this sequence}{input[i]}
+ *    If stride_ > 0:
+ *      Output: a shorten sequence. The operation of getting max instance of a
+ *              sequence is independently performed on every slice of the input
```
Comment (luotao1) [inline paddle/gserver/layers/MaxLayer.h]: Done. And update the comments in SequencePoolLayer.h, AverageLayer.h and SequenceLastInstanceLayer.h at the same time.

**Pull request merged**